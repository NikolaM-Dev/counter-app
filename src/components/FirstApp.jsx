import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ greeting, subtitle }) => {
  return (
    <>
      <h1>{greeting}</h1>
      <p>{subtitle}</p>
      <button>+ 1</button>
    </>
  );
};

FirstApp.propTypes = {
  greeting: PropTypes.string,
};

FirstApp.defaultProps = {
  subtitle: "I'm a subtitle",
};

export default FirstApp;
