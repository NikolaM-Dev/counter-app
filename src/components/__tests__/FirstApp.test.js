import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';

import FirstApp from '../FirstApp';

describe('Test in <FirstApp />', () => {
  // it('should to show the message "Hi, i\'m Nikola"', () => {
  //   const greeting = "Hi, i'm Goku";
  //   const { getByText } = render(<FirstApp greeting={greeting} />);
  //   expect(getByText(greeting)).toBeInTheDocument();
  // });

  const greeting = "Hi i'm Nikola";

  it('should to show <FirstApp/> correctly', () => {
    const wrapper = shallow(<FirstApp greeting={greeting} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should to show the subtitle sent by props', () => {
    const subtitle = 'Subtitle ðŸ˜€';
    const wrapper = shallow(
      <FirstApp greeting={greeting} subtitle={subtitle} />
    );
    const textParagraph = wrapper.find('p').text();

    expect(textParagraph).toBe(subtitle);
  });
});
