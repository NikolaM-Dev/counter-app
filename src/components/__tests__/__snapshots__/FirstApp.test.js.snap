// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FirstApp /> should to show <FirstApp/> correctly 1`] = `
<Fragment>
  <h1>
    Hi i'm Nikola
  </h1>
  <p>
    I'm a subtitle
  </p>
</Fragment>
`;
